internal/dispatch/caching/caching.go

```diff
@@ -407,10 +407,12 @@ func (cd *Dispatcher) DispatchLookupSubjects(req *v1.DispatchLookupSubjectsReque
 		for _, slice := range cachedResultRaw.([][]byte) {
 			var response v1.DispatchLookupSubjectsResponse
 			if err := response.UnmarshalVT(slice); err != nil {
-				return fmt.Errorf("could not publish cached lookup subjects result: %w", err)
+				return err
 			}
 			if err := stream.Publish(&response); err != nil {
-				return fmt.Errorf("could not publish cached lookup subjects result: %w", err)
+				// wrapping status.Status into an error means status.FromError()
```

Comment:
not sure I follow...

Reply:
let me try to explain it with a test-case. The assertion in this test fails:

```go
func TestStatusWontBeUnwrapped(t *testing.T) {
	err := status.Errorf(codes.NotFound, "%s", errors.New("test"))
	wrappedErr := fmt.Errorf("wrap your status error: %w", err)
	_, ok := status.FromError(wrappedErr)
	require.True(t, ok)
}

Reply:
another thing I explored was trying to do `errors.As` into a `status.Error` before calling `status.FromError`, but unfortunately it's internal gRPC API. Maybe you have a different idea?